manifests/config.pp

```diff
@@ -102,7 +102,7 @@
   # Just convert to yaml
   file {"${config_path}/${config_file}":
     ensure  => $ensure,
-    content => inline_template('<%= @cobbler_config.to_yaml %>'),
```

Comment:
The result of @cobbler_config.sort.to_h.to_yaml is not what you might expect in actual cobbler configuration yaml file. While sorting first level of keys with @cobbler_config.sort.to_h.to_yaml  the second level of params are still generated in random order.
The proper sorting solution ended up as inline recursive function in template.

As for the duplicate parameter I will push update to the code.
Also the travis is complaining about unexpected key/value returns in functionality tests (which was not part of my changes) and will need some fixing outside of my work.